spec/controllers/spree/api/line_items_controller_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+
+module Spree
+  describe Spree::Api::LineItemsController do
+    render_views
+
+    before do
+      stub_authentication!
+      Spree.user_class.stub :find_by_spree_api_key => current_api_user
+    end
+
+    def self.make_simple_data!
+      let!(:order) { FactoryGirl.create(:order, state: 'complete', completed_at: Time.now) }
+      let!(:line_item) { FactoryGirl.create(:line_item, order: order, unit_value: 500) }
+    end
+
+    #test that when a line item is updated, an order's fees are updated too
+    context "as an admin user" do
+      sign_in_as_admin!
+      make_simple_data!
+
+      context "as a line item is updated" do
+        it "apply enterprise fees on the item" do
+          line_item_params = { order_id: order.number, id: line_item.id, line_item: { id: line_item.id, unit_value: 520 }, format: :json}
+          controller.should_receive(:apply_enterprise_fees).and_return(true)
```

Comment:
Hi @lin-d-hop,

Yep, basically as soon as you stub a method with should_receive, it gets replaced with a spy which listens for calls, and no code within the method is run. You can allow a pass through by adding .and_call_original to the stub. Of course you can't add a return value when you use .and_call_original.

What I was trying to say in the original message is that I don't think we want to stub apply_enterprise_fees at all, because testing that it is just testing that rails is doing what rails should do. ie. calling a method specified in an after filter at the appropriate time. Testing that is not our responsibility. The behaviour we are actually interested in testing is that .update_distribution_charge! is called on the order after the update action.

So, I think you should remove the stub (should_receive) on the apply_enterprise_fees method on the controller, and instead stub the .update_distribution_charge! method on the order, and check that that method is called when you call the line_item update action.

Something like this:
order.should_receive(:update_distribution_charge!)

Or, if you wanted to use the newer RSpec syntax:
expect(order).to receive(:update_distribution_charge!)